p2p/discovery/mdns.go

```diff
@@ -131,18 +155,34 @@ func (m *mdnsService) pollForEntries(ctx context.Context) {
 			}
 		}()
 
-		log.Debug("starting mdns query")
+		// get interface from name, if any, and verify the interface is up before sending a new query
+		var iface *net.Interface
+		var err error
+		if m.ifaceName != "" {
+			iface, err = net.InterfaceByName(m.ifaceName)
```

Comment:
Shouldn't we do this at the top of this function and get `NewMdnsService ` to return an error if there is no such interface on the host ?

Reply:
I thought about this but then I wondered how it would behave if the interface is turned down during execution...
That's why I decided to get the interface in the loop to make sure the `iface` is always up to date.

What do you think? 

Regarding checking the interface name from `NewMdnsService`, I assume the interface exists when the function is called as a pointer is passed: `iface *net.Interface`. But that can indeed be added.

Reply:
@fouge 

IMO, when `NewMdnsService ` is called with a non nil interface and no such interface exists on the machine, we should probably return an error. 

Are you thinking of a scenario where a new interface would be added to the machine after MDNS starts ? That dosen't sound like it'd happen often.

Reply:
I agree about that point: when the service starts, the interface should exist. I have added a check when the service is created (in `NewMdnsService`) using the interface name, as it's now an interface name that is passed instead of a pointer to the interface (since I changed with optional functions to 

Should we keep the `iface` instantiation in the loop? 
I think this is more resilient.